connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -247,6 +248,19 @@ public boolean isSinkConnector(String connName) {
         }
     }
 
+    /*
+     * Retrieves the connector type by connector name
+     *
+     * @return the ConnectorType for the given connector name.
+     *  null if the worker does not manage a connector with the given name.
+     */
+    public ConnectorType getConnectorType(String connName) {
+        WorkerConnector workerConnector = connectors.get(connName);
+        if (workerConnector == null)
+            return null;
```

Comment:
Do we really want to return `null` here? This is only going to be used (at least currently) by the two REST APIs, right? If we can't even find the connector, it seems like a 404 or some other error should be happening since we aren't even aware of the connector? This might not be the right location to trigger the exact error, but presumably we should either use the `null` sentinel value to trigger an exception elsewhere or throw an exception here which is ultimately converted into the right exception to trigger an error status code?